ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -2543,7 +2531,7 @@ FEDataManager::applyGradientDetector(const Pointer<BasePatchHierarchy<NDIM> > hi
         // level.
         std::vector<std::vector<Elem*> > active_level_elem_map;
         const IntVector<NDIM> ghost_width = 1;
-        collectActivePatchElements(active_level_elem_map, level_number);
+        collectActivePatchElements(active_level_elem_map, level_number, level_number + 1, d_max_level_number);
```

Comment:
This more precise API call replaces `if (getPatchLevel(elem) <= level_number) continue;`